src/base-element.js

```diff
@@ -264,7 +264,10 @@ export class BaseElement {
    * @return {boolean}
    */
   renderOutsideViewport() {
-    return true;
+    const box = this.getIntersectionElementLayoutBox();
```

Comment:
Ideally I'd rather have this method to be very light and very declarative. Runtime generally would have a better idea on how to to calculate the right window. E.g. when we are scrolling down - the window could be made bigger in the direction of scrolling, etc. It'd be very hard to capture these improvements inside of this method.